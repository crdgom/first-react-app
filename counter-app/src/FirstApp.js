

const FirstApp = ({hello = 'Hello Word'})=>{
    
    return(
        <>
            <h1>{hello}</h1>
            {/* <h1>{JSON.stringify(hello, null, 3)}</h1> */}
            <p>Mi primera app</p>
        </>
    );
    
}

export default FirstApp