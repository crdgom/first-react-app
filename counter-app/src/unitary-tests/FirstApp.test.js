import React from 'react';
import { render } from '@testing-library/react';
import FirstApp from "../FirstApp";
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('FirstApp module Test',()=>{
    /* test('Should return h1 string (Hola, soy Carlos)',()=>{
        const hello = 'Hola, soy Carlos';
        const {getByText} = render(<FirstApp hello={hello}/>);
        expect(getByText(hello)).toBeInTheDocument();

    }) */
    test('First real test with enzyme test library',()=>{
        const hello = 'Hola, soy Carlos'
        const wrapper = shallow(<FirstApp hello = {hello}/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('must show the paragraph with the subtitle', ()=>{
        const hello = 'Hola, soy Carlos';
        const subtitle ='This is a subtitle'
        const wrapper = shallow(
            <FirstApp
                hello = {hello}
                subtitle={subtitle}
            />)
        const paragraph = wrapper.find('p').text() 
        expect(paragraph).toBe(subtitle);
    })
})