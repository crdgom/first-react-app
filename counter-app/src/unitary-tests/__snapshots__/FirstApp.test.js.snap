// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp module Test First real test with enzyme test library 1`] = `
<Fragment>
  <h1>
    Hola, soy Carlos
  </h1>
  <p>
    This is a subtitle
  </p>
</Fragment>
`;
